x-pack/plugins/actions/server/plugin.ts

```diff
@@ -179,7 +180,17 @@ export class ActionsPlugin implements Plugin<Promise<PluginSetupContract>, Plugi
     executeActionRoute(router, this.licenseState, actionExecutor);
 
     return {
-      registerType: actionTypeRegistry.register.bind(actionTypeRegistry),
+      registerType: (actionType: ActionType) => {
+        if (!(actionType.minimumLicenseRequired in LICENSE_TYPE)) {
+          throw new Error(`"${actionType.minimumLicenseRequired}" is not a valid license type`);
+        }
+        if (LICENSE_TYPE[actionType.minimumLicenseRequired] < LICENSE_TYPE.gold) {
```

Comment:
@elastic/kibana-alerting-services @peterschretlen do we want to allow other plugins to register `basic` license action types? With the code I have now, only built-in action types can specify basic as their minimum license required.

Reply:
@mikecote  👍 the way you have it is correct. Custom actions would be gold or higher.